compiler/rustc_resolve/src/ident.rs

```diff
@@ -866,10 +877,17 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
                     if ident.name == kw::Crate || ident.name == kw::DollarCrate {
                         let module = self.resolve_crate_root(ident);
                         return Ok(module.self_binding.unwrap());
-                    } else if ident.name == kw::Super || ident.name == kw::SelfLower {
-                        // FIXME: Implement these with renaming requirements so that e.g.
-                        // `use super;` doesn't work, but `use super as name;` does.
-                        // Fall through here to get an error from `early_resolve_...`.
+                    } else if ident.name == kw::Super {
+                        if let Some(parent) = parent_scope.module.parent {
+                            let mut ctxt = ident.span.ctxt().normalize_to_macros_2_0();
+                            return Ok(self.resolve_self(&mut ctxt, parent).self_binding.unwrap());
+                        }
```

Comment:
The logic for `super` here and above doesn't do the same thing as in `fn resolve_path_with_ribs`.
It would be better to factor it into a separate function and use it from all the 3 places.